lib/README.md

```diff
@@ -295,3 +295,110 @@ if (false === $result) {
 ```
 
 This function returns the error message if any.
+
+### Class `WasmArrayBuffer`
+
+This class represents a buffer of bytes. It will be used to manipulate
+the memory of a WebAssembly instance.
+
+The class looks like this:
+
+``` php
+class WasmArrayBuffer
+{
+    public function __construct(int $byte_length);
+    public function getByteLength(): int;
+}
+```
+
+### Classes `WasmTypedArray`
+
+`WasmTypedArray` is a generic name to represent classes that act as
+array-like views over a `WasmArrayBuffer`.
+
+| Class | View buffer as a sequence of… | Bytes per element |
+|-|-|-|
+| `WasmInt8Array` | `int8` | 1 |
+| `WasmUint8Array` | `uint8` | 1 |
+| `WasmInt16Array` | `int16` | 2 |
+| `WasmUint16Array` | `uint16` | 2 |
+| `WasmInt32Array` | `int32` | 4 |
+| `WasmUint32Array` | `uint32` | 4 |
+
+They all share the same implementation. Taking the example of
+`WasmUint8Array`, it looks like this:
+
+```php
+class WasmUint8Array implements ArrayAccess
+{
+    public const BYTES_PER_ELEMENT;
+    
+    public function __construct(WasmArrayBuffer $wasm_array_buffer, int $offset = 0, int $length = 0);
+    public function getOffset();
+    public function getLength();
+
+    /* For `ArrayAccess` */
+    public function offsetGet($offset): int;
+    public function offsetSet($offset, $value);
+    public function offsetExists($offset): bool;
+    public function offsetUnset($offset);
```

Comment:
`: void`

Reply:
I thought I can't, but actually it's possible. Code, tests & doc' updated :+1:.